akka-sample-sharding-scala/src/main/scala/sample/sharding/EdgesApp.scala

```diff
@@ -0,0 +1,163 @@
+ * taking samples and posting via HTTPS.
+ */
+object EdgesApp {
+
+  val WeatherStations = 50
+  // temperature, pressure, precipitation, wind speed...
+  val DevicesPerStation = 1
+
+  val random = new Random()
+
+  def main(args: Array[String]): Unit = {
+    val config =
+      ConfigFactory
+        .parseString(s"akka.remote.artery.canonical.port = 0")
+        .withFallback(ConfigFactory.load("weather-station.conf"))
+
+    // simple sample, using one node on local machine
+    val rootBehavior = Behaviors.setup[Nothing] { context =>
+      (0 to WeatherStations).foreach { sid =>
+        context.spawn(Station(sid, DevicesPerStation, 8080), s"station-$sid")
+      }
+
+      Behaviors.empty
+    }
+
+    ActorSystem[Nothing](rootBehavior, "SimulatedDevices", config)
+  }
+}
+
+object Station {
+
+  def apply(stationId: Int, devices: Int, port: Int): Behavior[Device.Batch] =
+    Behaviors.setup { context =>
+      context.log.info("Started station with {} devices.", devices)
+      val http = context.spawn[Device.Batch](Post(stationId, port), s"http-post-$stationId")
+
+      (0 to devices).foreach { deviceId =>
+        context.spawn(Device(stationId, deviceId, http, 5.second, 10), s"device-$deviceId")
+      }
+
+      Behaviors.empty
+    }
+}
+
+/**
+ * Any given weather station has a unique ID, and can sample temperature, wind speed, precipitation, etc.
+ * from multiple devices.
+ *
+ * For a simple example, this does one type of reading.
+ */
+object Post {
+
+  def apply(stationId: Int, port: Int): Behavior[Device.Batch] =
+    Behaviors.setup[Device.Batch] { context =>
+      import akka.http.scaladsl.Http
+      import akka.http.scaladsl.model.{ ContentTypes, HttpEntity, HttpMethods, HttpRequest }
+      import context.executionContext
+
+      import akka.actor.typed.scaladsl.adapter._
+      val http = Http(context.system.toClassic)
+
+      Behaviors.receiveMessage[Device.Batch] {
+        case Device.Batch(buffered) =>
+          context.log.info("Received batch: {}.", buffered)
+
+          // TODO marshaller
+          val data = buffered.map {
+            case Device.Data(a, b, c) =>
+              s"""{"stationId":$a,"deviceId":$b,"latest":$c}"""
+          }
+          val json = s"""{"stations": [${data.mkString(",")}]}"""
+          context.log.debug("Read: {}", json)
+
+          // See https://doc.akka.io/docs/akka-http/current/common/index.html
+          // for marshalling, unmarshalling, json support in the wild.
+          val httpEntity = HttpEntity(ContentTypes.`application/json`, json)
+          context.log.info("Sending {}.", httpEntity)
+
+          http
+            .singleRequest(
+              HttpRequest(method = HttpMethods.POST, uri = s"http://localhost:$port/temperature", entity = httpEntity))
+            .onComplete {
+              case Success(res) => context.log.info(res.toString())
+              case Failure(e)   => context.log.error("Something wrong.", e)
+            }
```

Comment:
Is there a better way?

Reply:
or use a standalone `Logger` from `LoggerFactory.getLogger` if it's not important to correlate the logging to the specific actor?

Reply:
The update now batches for a max buffered number of readings before posting. I'll update the logger, thanks.